lib/puppet_x/puppetlabs/registry/provider_base.rb

```diff
@@ -39,8 +38,35 @@ def self.define_ffi
     #   _Reserved_  DWORD Reserved
     # );
     ffi_lib :advapi32
-    attach_function :RegDeleteKeyExA,
+    attach_function :RegDeleteKeyExW,
       [:handle, :pointer, :win32_ulong, :dword], :win32_long
+
+    # TODO: this duplicates code found in puppet, but necessary for backwards compat
+    class << base
```

Comment:
On a base install trying to run `bundle exec rspec spec` ends in a failure of 
```
C:/repos/registry/lib/puppet_x/puppetlabs/registry/provider_base.rb:46:in `define_ffi': Could not autoload puppet/provider/registry_key/registry: unde
fined local variable or method `base' for PuppetX::Puppetlabs::Registry::ProviderBase:Module (Puppet::Error)
```
Commenting this out we succeed in running the test but expectedly we fail.
Should we be doing this or just completely calling an individual thing for that is not tied to Puppet::Provider base class